jbehave-support-core/src/main/java/org/jbehavesupport/core/internal/web/action/SelectWebAction.java

```diff
@@ -23,9 +23,17 @@ public void perform(WebActionContext ctx) {
 
         WebElement element = findElement(ctx);
 
-        Assert.isTrue(equalsIgnoreCase(element.getTagName(), "select"), "The SELECT action is not supported on this element type: " + element.getTagName());
-
-        new Select(element).selectByVisibleText(ctx.getData());
+        if (equalsIgnoreCase(element.getTagName(), "input") && equalsIgnoreCase(element.getAttribute("type"), "checkbox")){
+            boolean checked = ctx.getData().isEmpty() || toBoolean(ctx.getData());
```

Comment:
minor: I'd use some null-safe isEmpty util method to prevent NPE, there is no guarantee that getData method will return non-null value